test/src/main/java/org/apache/accumulo/test/functional/HalfDeadServerWatcherIT.java

```diff
@@ -0,0 +1,212 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(HalfDeadTabletServer.class);
+
+    public static void main(String[] args) throws Exception {
+      try (HalfDeadTabletServer tserver = new HalfDeadTabletServer(new ServerOpts(), args)) {
+        tserver.runServer();
+      }
+    }
+
+    public static class StuckWatcher implements Watcher {
+      private static final Logger LOG = LoggerFactory.getLogger(StuckWatcher.class);
+
+      @Override
+      public void process(WatchedEvent event) {
+        LOG.info("started sleeping...");
+        while (true) {
+          LOG.info("still sleeping...");
+          UtilWaitThread.sleep(2000);
+        }
+      }
+
+    }
+
+    protected HalfDeadTabletServer(ServerOpts opts, String[] args) {
+      super(opts, args);
+    }
+
+    @Override
+    protected TreeMap<KeyExtent,TabletData> splitTablet(Tablet tablet, byte[] splitPoint)
+        throws IOException {
+      LOG.info("In HalfDeadServerWatcherIT::splitTablet");
+      TreeMap<KeyExtent,TabletData> results = super.splitTablet(tablet, splitPoint);
+      if (!tablet.getExtent().isMeta()) {
+        final TableId tid = tablet.getExtent().tableId();
+        final String zooRoot = this.getContext().getZooKeeperRoot();
+        final String tableZPath = zooRoot + Constants.ZTABLES + "/" + tid.canonical();
+        try {
+          this.getContext().getZooReaderWriter().exists(tableZPath, new StuckWatcher());
+        } catch (KeeperException | InterruptedException e) {
+          LOG.error("Error setting watch at: {}", tableZPath, e);
+        }
+        LOG.info("Set StuckWatcher at: {}", tableZPath);
+      }
+      return results;
+    }
+  }
+
+  private static final AtomicBoolean USE_VERIFICATION_THREAD = new AtomicBoolean(false);
+
+  @Override
+  public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    if (USE_VERIFICATION_THREAD.get()) {
+      cfg.setProperty(Property.GENERAL_SERVER_LOCK_VERIFICATION_INTERVAL, "10s");
+    } else {
+      cfg.setProperty(Property.GENERAL_SERVER_LOCK_VERIFICATION_INTERVAL, "0");
+    }
+    cfg.setServerClass(ServerType.TABLET_SERVER, HalfDeadTabletServer.class);
+    cfg.setNumCompactors(0);
+    cfg.setNumScanServers(0);
+    cfg.setNumTservers(1);
+  }
+
+  @AfterEach
+  public void afterTest() throws Exception {
+    getCluster().getClusterControl().stopAllServers(ServerType.TABLET_SERVER);
+    super.teardownCluster();
+    USE_VERIFICATION_THREAD.set(!USE_VERIFICATION_THREAD.get());
+  }
+
+  @Test
+  public void testOne() throws Exception {
+    if (USE_VERIFICATION_THREAD.get()) {
+      // This test should use the verification thread, which should
+      // end the TabletServer, throw an Exception on the ping call,
+      // and return true
+      assertTrue(testTabletServerWithStuckWatcherDies());
+    } else {
+      // This test should time out
+      assertThrows(IllegalStateException.class, () -> testTabletServerWithStuckWatcherDies());
```

Comment:
What does the exception look like for this?  Wondering if anything more specific can be checked for the exception.  Is the expecation that this exception is coming from the call to `Wait.waitfor`?

Reply:
>  Is the expecation that this exception is coming from the call to Wait.waitfor?

yes, in the case that the server is still up, then pingServer will continue to return true and then Wait.waitFor will time out and throw an IllegalStateException. I could test that the message contains "Timeout exceeded".

Reply:
Updated the test in 601aff5. IT still passes.